### PR TITLE
No frontlight devices - disable irrelevant plugins

### LINKED ISSUE #12764: [Kindle w/o Front light] Unrelated plugins should be disabled
* KOReader version: 2024.07-285-gfd3727274 (legacy)
* Device: Kindle 4 Non-Touch (FW 4.1.4)

#### Issue
I believe these plugins should be disabled on non-front light Kindles (K2, DXG, K3, K4, Touch):
- `Auto frontlight (outdated)`
- `Auto night mode` (it says `This plugin allows to set the frontlight warmth automagically.` in the helper Info Message)
- `Automatic dimmer`

If you guys agree, I can work on a PR which will just add checks in `main.lua` of these plugins at the top, similarly to, say, gestures plugin (but with correct check):
```
if not Device:hasFrontlight() then
    return { disabled = true, }
end
```

#### Steps to reproduce
Open plugin management list.

##### `crash.log` (if applicable)
Not needed

### PR DESCRIPTION
**Pull request by mergen3107**
Closes #12764 

However, adding these to K4NT did not make them disappear from the list.

Hence, draft PR.

To be continued...

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12766)

### CODE REVIEW

**Comment by mergen3107** [comment]
AutoWarmth plugin's `_meta.lua` only checks for just warm light presence...
Maybe I can add frontlight check there.

**Comment by mergen3107** [comment]
Great point! Thank you, I'll try soon

File: plugins/autowarmth.koplugin/main.lua [hunk 1]
```diff
@@ -31,6 +31,10 @@ local T = FFIUtil.template
 local Screen = require("device").screen
 local datetime = require("datetime")
 
+if not Device:hasFrontlight() then
+    return { disabled = true, }
+end
```
Comment (Frenzie) [inline plugins/autowarmth.koplugin/main.lua:36]: It can also do auto-nightmode.

File: plugins/autofrontlight.koplugin/main.lua [hunk 2]
```diff
@@ -4,6 +4,10 @@ if not (Device:isKindle() and Device:hasLightSensor()) then
     return { disabled = true, }
 end
 
+if not Device:hasFrontlight() then
+    return { disabled = true, }
+end
+
```
Comment (Commodore64user) [inline plugins/autofrontlight.koplugin/main.lua:10]: this one seems unnecessary, there is the `not (Device:isKindle() and Device:hasLightSensor())` check just above it and every kindle with light sensor already has a front light... and is being deprecated and removed soon nonetheless.

Comment (mergen3107) [reply plugins/autofrontlight.koplugin/main.lua:10]: Do you know why I still see it in the list?

Or, am I confusing it thinking that "disabled" should remove it from the list?

**Review by Frenzie — APPROVED** [review]
lgtm, didn't test

**Comment by mergen3107** [comment]
@Frenzie 
Please wait, I yet need to figure out why on my K4NT these plugins aren't disabled with this PR

**Comment by mergen3107** [comment]
Oh, that's weird.

I guess I found a culprit!

It is... (please don't beat me up xD) ... my transferred settings from Kindle Scribe!

The way I figured it out is that I enabled `Autofronlight` plugin and restarted.

After restart, KOReader kicked `Autofrontlight` out of the plugin list :D 

So the "list" must have been fetched as is from `settings.reader.lua`, but never checked again upon actual usage.

Can we fix that too?

**Comment by mergen3107** [comment]
@Frenzie 
This PR is ready for review!

I found out why it didn't "work" for me, see my post above.

> So the "list" must have been fetched as is from settings.reader.lua, but never checked again upon actual usage.
Can we fix that too?

This can wait until another PR, if anyone knows a solution.

Thank you!

**Comment by Frenzie** [comment]
> This can wait until another PR

Doesn't really belong in this one regardless. :-)

File: plugins/autowarmth.koplugin/_meta.lua [hunk 3]
```diff
@@ -2,5 +2,5 @@ local _ = require("gettext")
 return {
     name = "autowarmth",
     fullname = require("device"):hasNaturalLight() and _("Auto warmth and night mode") or _("Auto night mode"),
-    description = _([[This plugin allows to set the frontlight warmth automagically.]]),
+    description = require("device"):hasNaturalLight() and _([[This plugin allows to set the frontlight warmth and night mode automagically.]]) or _([[This plugin allows to set night mode automagically.]]),
```
Comment (Frenzie) [inline plugins/autowarmth.koplugin/_meta.lua:5]: It's not really set, is it.
```suggestion
    description = require("device"):hasNaturalLight() and _([[This plugin allows to set the frontlight warmth and night mode automagically.]]) or _([[This plugin allows to enable night mode automagically.]]),
```

**Comment by mergen3107** [comment]
I synced this branch to prepare for merging.
I hope I didn't break it :D

**Pull request merged**